### ISSUE TITLE
Amazon S3 bcbio_vm AMI

### ISSUE THREAD

**Issue by mortunco** [chapmanb/bcbio-nextgen-vm#142]
Hi,

I have seen that there is a bcbio_vm AMI under community AMIs under launching instances. What would be the benefit of starting an bcbio_vm AMI? Would there be a benefit?

Also, for processing data in the buckets should I install bcbio to the buckets everytime or downloading to my root will be enough?

Thank you.

Best,

Tunc.

**Comment by chapmanb** [comment]
Tunc;
The community AMI had a pre-installed version of bcbio + data for the teaching example documentation:

https://bcbio-nextgen.readthedocs.org/en/latest/contents/teaching.html

I'd only use this if you're following along with that example.

I'm not sure exactly what you're asking with regards to S3 buckets. Could you provide more details so we can try to help? Thanks much.

**Comment by mortunco** [comment]
Oh I did not know that community AMI could solve all the issues with downloading. Posting this could be an useful tip.

Let me rephrase my second question. Like I mentioned before, I am aiming to process bam files in the ICGC repository. ( which collaborates with amazon to keep all the files in a Amazon S3 bucket). So my question is; instead of downloading bam files in to an EBS volume, Can I directly process the information from icgc buckets? For example; input directory will be ICGC amazon S3 bucket, output directory is my amazon ec2 virtual machine ?

Thank you very much for your time,

Best,

Tunc.

**Comment by chapmanb** [comment]
Tunc;
Thanks for the clarifications. Regarding the community AMI, I'd only use that for replicating the tutorial since it's not being actively updated (until we revise the example) and is not meant as a production deployment.

You can use files in other S3 buckets and you'll need to reference them manually as inputs in your sample YAML file:
```
  files: s3://bucket/file_name.bam
```
bcbio will download them locally and do processing on the local EBS volume on your machine as part of the run. Hope this helps.
